src/pages/add-products/components/product-category-and-type/index.tsx

```diff
@@ -0,0 +1,217 @@
+		(state: RootState) => state.productForm.productCategory,
+	);
+
+	const { data: categoriesWithTypes } =
+		useGetCategoriesWithTypesQuery(undefined);
+	const { data: styles } = useGetProductStylesQuery(undefined);
+
+	const [selectedCategory, setSelectedCategory] = useState<null | number>(
+		category,
+	);
+	const [selectedType, setSelectedType] = useState<null | number>(type);
+	const [selectedStyle, setSelectedStyle] = useState<null | number>(style);
+
+	const [types, setTypes] = useState<OptionType[]>([]);
+
+	const { errors, handleSubmit, setValue } =
+		useAppForm<ProductCategoryTypeStyle>({
+			defaultValues: {
+				category: null,
+				style: null,
+				type: null,
+			},
+			validationSchema: categoryTypeStyleValidation,
+		});
+
+	useEffect(() => {
+		setValue("category", category);
+		setValue("style", style);
+		setValue("type", type);
+	}, [category, style, type, setValue]);
+
+	useEffect(() => {
+		if (categoriesWithTypes && selectedCategory) {
+			const category = categoriesWithTypes.find(
+				(category) => category.id === selectedCategory,
+			);
+
+			if (category) {
+				const typesOptions: OptionType[] = category.types.map((type) => ({
+					label: type.name,
+					value: type.id,
+				}));
+				setTypes(typesOptions);
+			}
+		}
+	}, [categoriesWithTypes, selectedCategory]);
+
+	const handleInputChange = useCallback(
+		async (formData: ProductCategoryTypeStyle): Promise<void> => {
+			dispatch(updateProductCategoryTypeStyle(formData));
+			navigate(AppRoute.PRODUCT_DESCRIPTION);
+		},
+		[dispatch, navigate],
+	);
+
+	const handleCategoryChange = useCallback(
+		(event: SelectChangeEvent<number>) => {
+			const categoryId = +event.target.value;
+			setSelectedCategory(categoryId);
+			setValue("category", categoryId);
+		},
+		[setValue],
+	);
+
+	const handleTypeChange = useCallback(
+		(event: SelectChangeEvent<number>) => {
+			const typeId = +event.target.value;
+			setSelectedType(typeId);
+			setValue("type", +event.target.value);
+		},
+		[setValue],
+	);
+
+	const handleStyleChange = useCallback(
+		(event: SelectChangeEvent<number>) => {
+			const styleId = +event.target.value;
+			setSelectedStyle(styleId);
+			setValue("style", +event.target.value);
+		},
+		[setValue],
+	);
+
+	const handleFormSubmit = useCallback(
+		(event_: React.BaseSyntheticEvent): void => {
+			event_.preventDefault();
+			void handleSubmit(handleInputChange)(event_);
+		},
+		[handleSubmit, handleInputChange],
+	);
+
+	const categoryOptions: OptionType[] = categoriesWithTypes
+		? categoriesWithTypes.map((category) => ({
+				label: category.name,
+				value: category.id,
+		  }))
+		: [];
+
+	const styleOptions: OptionType[] = styles
+		? styles.map((style) => ({
+				label: style.name,
+				value: style.id,
+		  }))
+		: [];
+
+	if (!categoriesWithTypes || !styles || !types) {
+		return (
+			<Box alignItems="center" display="flex" minHeight="550px">
+				<Loader />
+			</Box>
+		);
+	}
```

Comment:
I think its better to create wrapper component, where you will pass show/not show variable, and if you pass "show" it will render loader